pkg/ice/agent.go

```diff
@@ -192,10 +192,14 @@ func (a *Agent) sendSTUN(msg *stun.Message, local, remote Candidate) {
 }
 
 func (a *Agent) updateConnectionState(newState ConnectionState) {
-	a.connectionState = newState
```

Comment:
Can we remove this conditional now? https://github.com/pions/webrtc/blob/master/rtcpeerconnection.go#L786

I added that so the agent wouldn't spam the public API. But since the agent is doing that check now, we should be able to kill in `RTCPeerConnection`

thanks!